libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -410,6 +410,43 @@ bool ExpressionCompiler::visit(TupleExpression const& _tuple)
 bool ExpressionCompiler::visit(UnaryOperation const& _unaryOperation)
 {
 	CompilerContext::LocationSetter locationSetter(m_context, _unaryOperation);
+
+	if (FunctionDefinition const* function = _unaryOperation.annotation().userDefinedFunction)
+	{
+		FunctionType const* functionType = dynamic_cast<FunctionType const*>(
+			function->libraryFunction() ? function->typeViaContractName() : function->type());
+
+		solAssert(functionType);
+		functionType = dynamic_cast<FunctionType const&>(*functionType).asBoundFunction();
+		solAssert(functionType);
+		evmasm::AssemblyItem returnLabel = m_context.pushNewTag();
+		_unaryOperation.subExpression().accept(*this);
```

Comment:
Type conversion is missing here!
Can you add a test that exposes this? Maybe this is hard to find because of missing implicit conversion for user-defined types.

Reply:
As discussed it would be rather hard to provide a test for that.